src/app/clusters/ota-requestor/BDXDownloader.cpp

```diff
@@ -103,6 +103,18 @@ CHIP_ERROR BDXDownloader::FetchNextData()
     return CHIP_NO_ERROR;
 }
 
+uint8_t BDXDownloader::GetDownloadPercentComplete()
+{
+    if (mBdxTransfer.GetTransferLength() == 0)
+    {
+        return 0;
+    }
+    else
+    {
+        return static_cast<uint8_t>((mBdxTransfer.GetNumBytesProcessed() * 100) / mBdxTransfer.GetTransferLength());
+    }
+}
```

Comment:
It is possible that there is no definite length available for the transfer. Spec makes clear that in that case, null must be used. Should have Optional<uint8_t> here if percentage is not available.

Reply:
FYI, I just recently made a change `GetPercentComplete` in OTAImageProcessor.h in https://github.com/project-chip/connectedhomeip/pull/13727 which addressed this.

Reply:
I guess that was already addressed by @carol-apple